src/main/java/htsjdk/variant/variantcontext/GenotypeNumLikelihoodsCache.java

```diff
@@ -0,0 +1,43 @@
+package htsjdk.variant.variantcontext;
+
+import java.util.HashMap;
+
+/**
+ * Hybrid dynamic cache for genotype likelihood counts
+ */
+
+public class GenotypeNumLikelihoodsCache {
+
+    private final static int N_ALLELES = 5;
+    private final static int PLOIDY = 10;
+
+    private final int[][] staticCache = new int[5][10];
+    private final HashMap<Integer, HashMap<Integer, Integer>> dynamicCache = new HashMap<>();
```

Comment:
It would probably be faster to use a compound key and a single layer of HashMap, but it would be more complicated if we wanted to avoid creating a new key object on every lookup.

Reply:
I'm on board with using a compound key